### PR TITLE
Handle ask for optional password capability

### PR DESCRIPTION
**Pull request by er-vin**

### CODE REVIEW

File: src/gui/sharedialog.cpp [hunk 1]
```diff
@@ -33,6 +34,20 @@
 #include <QPushButton>
 #include <QFrame>
 
+namespace {
+QString createRandomPassword()
+{
+    const auto words = OCC::WordList::getRandomWords(10);
```
Comment (DominiqueFuchs) [inline src/gui/sharedialog.cpp:40]: Why not passing length as argument and use it with a default of 10 from the call in createLinkShare()? Could be used in a possibly more advanced manner later without modifying any of these then.

Comment (er-vin) [reply src/gui/sharedialog.cpp:40]: Well, I'd say YAGNI. ;-)

But that's also a question of leaking details. Indeed, that'd make the caller level of abstraction slightly lower due to the extra detail needing to be provided while now it has a clear "should create a random password? yes/no" mandate without having to bother about how this password is done.

Comment (DominiqueFuchs) [reply src/gui/sharedialog.cpp:40]: > YAGNI

Probably true, convinced

**Review by camilasan — APPROVED** [review]

**Review by misch7 — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Noice

**Pull request merged**